addons/sale/sale_analytic.py

```diff
@@ -13,7 +13,10 @@ def _compute_analytic(self, domain=None):
         lines = {}
         if not domain:
             # To filter on analyic lines linked to an expense
-            domain = [('so_line', 'in', self.ids), ('amount', '<=', 0.0)]
+            alines = self.env['account.analytic.line'].search([('so_line', 'in', self.ids), ('amount', '<=', 0.0)])
+            alines = alines.filtered(lambda l: l.amount > 0 or l.move_id == False or l.move_id.account_id.user_type_id.id == self.env.ref('account.data_account_type_expenses').id)
```

Comment:
in fact amount < 0, this is a distraction. For move_id == False, it s to keep the previous behavior for analytic line with amount = 0 and not linked to move_id